Add documentation for exposing submission API (#793)
Avaiga/taipy-doc: docs/manuals/core/entities/orchestrating-and-job-execution.md

In this section, it is assumed that <a href="./code_example/my_config.py" download>`my_config.py`</a>
module contains a Taipy configuration already implemented.

> To execute a scenario, you need to call the `taipy.submit()^` method. It returns a `Submission^` containing

Reviewer: ```suggestion
To execute a scenario, you need to call the `taipy.submit()^` function. It returns a `Submission^` object containing
```

import taipy as tp
import my_config

tp.Core().run()

scenario = tp.create_scenario(my_config.monthly_scenario_cfg)

tp.submit(scenario)

> tp.Core().run()

Reviewer 2: Why changing that? It is the opposite of what is described above.

Author: In the doc above: "Note that jobs can only be executed after the Taipy Core service is started." => The `tp.Core().run()` must be run first before we submit the scenario, the previous code was incorrect in that sense, no?

Reviewer 2: Not really. The Core service can be run before or after submitting jobs. 
But the execution starts only when the service is started.

These lines are here to show how to submit jobs before running the service.

taipy waits until all the submitted jobs are finished, or the timeout expires (which ever occurred
first).

> It is also possible to submit a single sequence using the same `taipy.submit()^` method. It returns

Reviewer: ```suggestion
It is also possible to submit a single sequence using the same `taipy.submit()^` function. It returns
```

> a `Submission^` containing the information about the submission of the sequence such as the created `Job^`s

Reviewer: ```suggestion
a `Submission^` object containing the information about the submission of the sequence such as the created `Job^`s
```

    sequence.submit()
    ```

> You can also submit a single task with the same `taipy.submit()^` method. It returns a `Submission^` containing

Reviewer: ```suggestion
You can also submit a single task with the same `taipy.submit()^` function. It returns a `Submission^` object containing
```

scenario = tp.create_scenario(my_config.monthly_scenario_cfg)
task = scenario.predicting

> submission = tp.submit(task)

Reviewer 2: Can we add 
```
if __name__ == "__main__":
```

    task.submit()
    ```

# Submission

Each time a `Scenario^`, a `Sequence^` or a `Task^` is submitted, a new `Submission^` entity is created.

## Submission attributes

Here is the list of the `Submission^`'s attributes:

> - _entity_id_: The identifier of the entity that was submitted.

Reviewer: ```suggestion
- *entity_id*: The identifier of the entity that was submitted.
```
and in the list below - use asterisks instead of underscore to avoid parsing problems with Python identifiers

- _id_: The identifier of the `Submission^` entity.
- _jobs_: A list of jobs.
- _properties_: A dictionary of additional properties.
- _creation_date_: The date of this submission's creation.
- _submission_status_: The current status of this submission.
- _version_: The string indicates the application version of the submission to instantiate.
  If not provided, the latest version is used.

## Submission Status

- `SUBMITTED`: The submission has been submitted for execution but not processed yet by the orchestrator.
- `UNDEFINED`: The submission's jobs have been submitted for execution but got some undefined status changes.

> - `PENDING`: The submission has been enqueued by the orchestrator. It is waiting for an executor to be available

Reviewer 2: ```suggestion
- `PENDING`: The orchestrator has enqueued the submission. It is waiting for an available worker to start executing a first job.
```

   for its execution.

> - `BLOCKED`: The submission has been blocked because it has been finished with a job being blocked.

Reviewer 2: ```suggestion
- `BLOCKED`: The submission is blocked by at least one blocked job waiting for its input data nodes to be ready.
```

- `RUNNING`: The submission has its jobs currently being executed.
- `CANCELED`: The submission has been submitted but its execution has been canceled.

> - `FAILED`: The submission has a job failed during its execution.

Reviewer: ```suggestion
- `FAILED`: The submission has a job that failed during its execution.
```

- `COMPLETED`: The submission has successfully been executed.

## Get/Delete Submission

> `Submission^` is created when a `Scenario^`, a `Sequence^` or a `Task^` is submitted.

Reviewer: ```suggestion
A `Submission^` object is created when a `Scenario^`, a `Sequence^` or a `Task^` is submitted.
```

> - You can get the latest submission of a `Scenario^`, a `Sequence^` or a `Task^` with `taipy.get_latest_job()^`.

Reviewer 2: ```suggestion
- You can get the latest submission of a `Scenario^`, a `Sequence^` or a `Task^` with `taipy.get_latest_submission()^`.
```

Author: ah hah thanks!!!

> - You can retrieve a `Submission^` from its id by using the `taipy.get()^` method.

Reviewer: ```suggestion
- You can retrieve a `Submission^` from its id by using the `taipy.get()^` function.
```

> A Submission can be deleted using the `taipy.delete()^` method.

Reviewer: ```suggestion
A Submission can be deleted using the `taipy.delete()^` function.
```

> Deleting a Submission can raise an `SubmissionNotDeletedException^` if the `Status^` of the Submission is not `CANCELED`,

Reviewer: ```suggestion
Deleting a Submission can raise a `SubmissionNotDeletedException^` if the `Status^` of the Submission is not `CANCELED`,
```

`COMPLETED`, `FAILED` or `UNDEFINED`.

!!! example

    ```python linenums="1"
    import taipy as tp

    def double(nb):
        return nb * 2

    print(f'(1) Number of submission: {len(tp.get_submissions())}.')

    # Create a scenario then submit it.
    input_data_node_config = tp.configure_data_node("input", default_data=21)
    output_data_node_config = tp.configure_data_node("output")
    task_config = tp.configure_task("double_task", double)
    scenario_config = tp.configure_scenario("my_scenario", [task_config])

    tp.Core().run()

    scenario = tp.create_scenario(scenario_config)
    tp.submit(scenario)

    # Retrieve all submission.
    print(f'(2) Number of submissions: {len(tp.get_submissions())}.')

    # Get the latest created submission of the scenario.
    tp.get_latest_submission(scenario)

    # Then delete it.
    tp.delete(submission)

>     print(f'(3) Number of submissions: {len(tp.get_submissions())}.')

Reviewer 2: Can we set this under a 
```
if __name__ == "__main__":
```

Author: I think we can but I don't think it's the convention in `taipy-doc`?

Reviewer 2: No it is not. But I am thinking of adding it, actually.
What do you think?

Author: haha it makes sense to! But in that case, then change should be applied to everywhere in the doc for consistency. Should we keep it in a separate PR?

Reviewer 2: OK for me.
